page-optimize.php

```diff
@@ -265,38 +269,65 @@ function page_optimize_schedule_cache_cleanup() {
 }
 
 // Cases when we don't want to concat
-function page_optimize_bail() {
+function page_optimize_should_optimize() {
 	// Bail if we're in customizer
 	global $wp_customize;
 	if ( isset( $wp_customize ) ) {
-		return true;
+		return false;
+	}
+
+	// Avoid interfering with AMP mode
+	if ( function_exists( 'amp_is_request' ) && amp_is_request() ) {
+		return false;
 	}
 
 	// Bail if Divi theme is active, and we're in the Divi Front End Builder
 	if ( ! empty( $_GET['et_fb'] ) && 'Divi' === wp_get_theme()->get_template() ) {
-		return true;
+		return false;
 	}
 
 	// Bail if we're editing pages in Brizy Editor
 	if ( class_exists( 'Brizy_Editor' ) && method_exists( 'Brizy_Editor', 'prefix' ) && ( isset( $_GET[ Brizy_Editor::prefix( '-edit-iframe' ) ] ) || isset( $_GET[ Brizy_Editor::prefix( '-edit' ) ] ) ) ) {
-		return true;
+		return false;
 	}
 
-	return false;
+	return true;
 }
 
 function page_optimize_init() {
-	if ( page_optimize_bail() ) {
+	if ( ! apply_filters( 'page_optimize_should_optimize' , page_optimize_should_optimize() ) ) {
 		return;
 	}
 
 	page_optimize_schedule_cache_cleanup();
 
-	require_once __DIR__ . '/settings.php';
-	require_once __DIR__ . '/concat-css.php';
-	require_once __DIR__ . '/concat-js.php';
+	if ( ! defined( 'ALLOW_GZIP_COMPRESSION' ) ) {
+		define( 'ALLOW_GZIP_COMPRESSION', true );
+	}
+
+	$concat_js = ! is_admin() && ( page_optimize_should_concat_js() || page_optimize_load_mode_js() );
+	if ( $concat_js ) {
+		require_once __DIR__ . '/concat-js.php';
+		global $wp_scripts;
+		$wp_scripts = new Page_Optimize_JS_Concat( $wp_scripts );
+		$wp_scripts->allow_gzip_compression = ALLOW_GZIP_COMPRESSION;
+	}
 
-	// Disable Jetpack photon-cdn for static JS/CSS
-	add_filter( 'jetpack_force_disable_site_accelerator', '__return_true' );
+	$concat_css = page_optimize_should_concat_css();
+	if ( $concat_css ) {
+		require_once __DIR__ . '/concat-css.php';
+		global $wp_styles;
+		$wp_styles = new Page_Optimize_CSS_Concat( $wp_styles );
+		$wp_styles->allow_gzip_compression = ALLOW_GZIP_COMPRESSION;
+	}
+
+	if ( $concat_js || $concat_css ) {
+		// Disable Jetpack photon-cdn for static JS/CSS if we're doing any concat
+		add_filter( 'jetpack_force_disable_site_accelerator', '__return_true' );
```

Comment:
If page-optimize isn't doing anything, we should probably let Jetpack site accelerator do it's thing. 😅